luaui/Widgets/gui_attackrange_gl4.lua

```diff
@@ -935,6 +974,52 @@ local function DrawBuilders()
 	end
 end
 
+local function updateScalingRange()			-- This function, and anything related to unitRangeScale, scalingUnitParams, and newRange are parts of a hook for Gunslingers or future/modded units that gain range with EXP.
+	local scalingUnitParams = unitRangeScale
+	for unitID, _  in pairs(scalingUnitParams.selections) do
+		local newRange = {}
+		local update
+		local scalingUnit = {}
+		if scalingUnitParams.selections[unitID] then 
+		scalingUnit = scalingUnitParams.selections[unitID]
+		end
+		for instanceID, j in pairs(scalingUnit.oldRange) do
+			local oldRange = scalingUnit.oldRange[instanceID]
+			local weaponNum = oldRange[1]
+			newRange[weaponNum] = spGetUnitWeaponState(unitID, weaponNum, "range")
+			if oldRange[2] ~= newRange[weaponNum] then
+				update = true
+			end
+		end
+		if update == true then
+			RemoveSelectedUnit(unitID, mouseover)			-- need to refresh this unit's ring
+			AddSelectedUnit(unitID, mouseover, newRange)
+		end
+	end
+	for unitID, _  in pairs(scalingUnitParams.mouseovers) do
+		local newRange = {}
+		local update
+		local mouseover
+		local scalingUnit = {}
+		if scalingUnitParams.mouseovers[unitID] then 
+		scalingUnit = scalingUnitParams.mouseovers[unitID]
+		end
+		for instanceID, j in pairs(scalingUnit.oldRange) do
+			local oldRange = scalingUnit.oldRange[instanceID]
+			local weaponNum = oldRange[1]
+			newRange[weaponNum] = spGetUnitWeaponState(unitID, weaponNum, "range")
+			if oldRange[2] ~= newRange[weaponNum] then
+				update = true
+				mouseover = true
+			end
+		end
+		if update == true then
```

Comment:
nit: explicit `== true` suggests that there's a meaningful distinction between the value being specifically of type bool rather than just generally truthy
```suggestion
		if update then
```